app/javascript/hyacinth_ui_v1/components/ui/forms/buttons/SubmitButton.js

```diff
@@ -21,7 +21,7 @@ class SubmitButton extends React.Component {
   }
 
   render() {
-    const { formType, saveData, ...rest } = this.props;
+    const { formType, saveData, staticContext, ...rest } = this.props;
```

Comment:
Why did you add staticContext here?

Reply:
If I don't, it gets pulled into `...rest`, and then gets passed along to another component and creates a warning about unexpected property `staticContext`.

Reply:
I have seen this error and I haven't had time to figure out a better solution. This works for now, but it will continue to come up, maybe we should ticket this? or at least put a comment right above?

Reply:
I don't know if it's necessary to ticket or comment because I think this is just a general coding pattern that we'll need to watch out for when using `...rest`, as it grabs a whole bunch of things that we don't always want to pass along to various sub-components.  We should just be careful about only passing down exactly what we need to various child components.  It'll prevent warnings and will also make PropType validation more useful, so we can trust that the documented prop types are the only ones actually being received.

Mentioning @barmintor @fcd1 @JackBlackLight here so they're all aware of this too.

Reply:
I though this problem only exists when we use the `withRouter` HOC component. When we move to functional components, we will use hooks to pull `history` and `params` from react-router and will no longer have a need for `withRouter` and I am hoping that removes this problem. That's why I think we should comment, or else we might continue to deconstruct a prop we don't actually have to deconstruct.

Reply:
Oh okay.  So you're concerned specifically about the staticContext prop?  I can add a comment about that.  I just meant that use of `...rest` in a similar way anywhere in the app is something to watch out for, if passing unnecessary props.

Reply:
Just added the comment we talked about.